### PR TITLE
Change from string to byte[] + headers

### LINKED ISSUE #5: Change from string to byte[]+headers

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes #5

### CODE REVIEW

File: fntypes.proto [hunk 1]
```diff
@@ -14,21 +14,14 @@
...
-    oneof type {
-	    string body = 1;
-	    google.protobuf.Empty void = 2;
+message Message {
+	message HeaderValue {
+		repeated string values = 1;
 	}
+
+    bytes payload = 1;
+	map<string, HeaderValue> headers = 2;
```
Comment (dturanski) [inline fntypes.proto:26]: fix indentation

File: fntypes.proto [hunk 2]
```diff
@@ -14,21 +14,14 @@
...
- * TODO: Reply can be Empty for sink behavior. Maybe we should create a separate RPC for sink.
- * For now the Dispatcher can check the return type and decide to output the Reply.
- */
-message Reply {
-    oneof type {
-	    string body = 1;
-	    google.protobuf.Empty void = 2;
+message Message {
+	message HeaderValue {
+		repeated string values = 1;
```
Comment (dturanski) [inline fntypes.proto:22]: why `repeated` here?

Comment (ericbottard) [reply fntypes.proto:22]: Because we want a `MultiMap<String, String>`

Comment (markfisher) [reply fntypes.proto:22]: @ericbottard decided we should represent multi-value headers explicitly (like HTTP)

Comment (dturanski) [reply fntypes.proto:22]: LGTM;

**Pull request merged**